test/metabase/lib/binning_test.cljc

```diff
@@ -65,3 +65,24 @@
       (is (=?
             {:metabase.lib.field/temporal-unit :minute}
             (m/find-first (comp #{"CREATED_AT"} :name) (lib/returned-columns query)))))))
+
+(deftest ^:parallel binning-equality-test
+  (testing "should compare 'default' binning values"
+    (are [expected x y] (= expected (lib.binning/binning= x y) (lib.binning/binning= y x))
+      true {:strategy :default} {:strategy :default}
+      false {:strategy :default} {:strategy :num-bins :num-bins 10}
+      false {:strategy :default} {:strategy :bin-width :bin-width 10}))
+  (testing "should compare 'num-bins' binning values"
+    (are [expected x y] (= expected (lib.binning/binning= x y) (lib.binning/binning= y x))
+      true {:strategy :num-bins :num-bins 10} {:strategy :num-bins :num-bins 10}
+      true {:strategy :num-bins :num-bins 10 :bin-width 3} {:strategy :num-bins :num-bins 10}
+      true {:strategy :num-bins :num-bins 10 :bin-width 3} {:strategy :num-bins :num-bins 10 :bin-width 4}
+      true {:strategy :num-bins :num-bins 10} {:strategy :num-bins :num-bins 10 :metadata-fn (fn [] nil)}
+      false {:strategy :num-bins :num-bins 10} {:strategy :num-bins :num-bins 20}))
+  (testing "should compare 'bin-width' binning values"
+    (are [expected x y] (= expected (lib.binning/binning= x y) (lib.binning/binning= y x))
+      true {:strategy :bin-width :bin-width 10} {:strategy :bin-width :bin-width 10}
+      true {:strategy :bin-width :bin-width 10 :num-bins 3} {:strategy :bin-width :bin-width 10}
+      true {:strategy :bin-width :bin-width 10 :num-bins 3} {:strategy :bin-width :bin-width 10 :num-bins 4}
+      true {:strategy :bin-width :bin-width 10} {:strategy :bin-width :bin-width 10 :metadata-fn (fn [] nil)}
+      false {:strategy :bin-width :bin-width 10} {:strategy :bin-width :bin-width 20})))
```

Comment:
<!--__GRAPHITE_HTML_TAG_START__--><i class='graphite__hidden'>[Re: lines +71 to +89]<br/><br/></i>

<!--__GRAPHITE_HTML_TAG_END__-->Generally we follow the convention not to use commas except for separating the key-value pairs of a one-line map, see the [style guide](https://github.com/bbatsov/clojure-style-guide?tab=readme-ov-file#optional-commas-in-map-literals).<!--__GRAPHITE_HTML_TAG_START__--><span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/metabase/metabase/46979?utm_source=unchanged-line-comment">Graphite</a>.</span>

<!--__GRAPHITE_HTML_TAG_END__-->

Reply:
Also I think this would be easier to read if the three columns were aligned.

Reply:
Fixed